### ISSUE TITLE
Failed to set state for sensor.metservice_metservice_weather_warnings

### ISSUE THREAD

**Issue by Greminn** [ciejer/metservice-weather#59]
## System Information

version | core-2024.4.2
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 6.6.20-haos
arch | x86_64
timezone | Pacific/Auckland
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1400
Downloaded Repositories | 23

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | April 30, 2024 at 12:00 PM
relayer_connected | true
relayer_region | ap-southeast-1
remote_enabled | true
remote_connected | true
alexa_enabled | false
google_enabled | false
remote_server | ap-southeast-1-2.ui.nabu.casa
certificate_status | ready
instance_id | 8e1e0743226e499f84b963e8948c79ba
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 12.1
-- | --
update_channel | stable
supervisor_version | supervisor-2024.04.0
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 30.8 GB
disk_used | 15.9 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.8.0), Log Viewer (0.17.0), Home Assistant Google Drive Backup (0.112.1), Silicon Labs Multiprotocol (2.4.5), Z-Wave JS (0.4.5), Mosquitto broker (6.4.0), Glances (0.21.1), Samba share (12.3.1), RTSPtoWebRTC (1.0.1), Advanced SSH & Web Terminal (17.2.0), Ring-MQTT with Video Streaming (5.6.3), openWakeWord (1.10.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 7
-- | --
resources | 15
views | 13
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | March 31, 2024 at 4:59 PM
-- | --
current_recorder_run | April 9, 2024 at 5:47 PM
estimated_db_size | 1637.82 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

### Checklist

- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/ciejer/metservice_weather/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Error in the home-assistant.log as per log below. 

PS: This is a really cool integration for HA. Well done in sorting it!!! 

### Reproduction steps

1.
2.
3.
...

### Debug logs

```text
2024-04-11 21:14:24.160 ERROR (MainThread) [homeassistant.helpers.entity] Failed to set state for sensor.metservice_metservice_weather_warnings, fall back to unknown
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1181, in _async_write_ha_state
    hass.states.async_set(
  File "/usr/src/homeassistant/homeassistant/core.py", line 2167, in async_set
    state = State(
            ^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 1654, in __init__
    validate_state(state)
  File "/usr/src/homeassistant/homeassistant/core.py", line 222, in validate_state
    raise InvalidStateError(
homeassistant.exceptions.InvalidStateError: Invalid state with length 1128. State max length is 255 characters.
```

**Comment by Greminn** [comment]
Ive got this as well... 

```
2024-05-20 20:35:34.063 ERROR (MainThread) [homeassistant.helpers.entity] Failed to set state for sensor.metservice_metservice_weather_warnings, fall back to unknown
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1203, in _async_write_ha_state
    hass.states.async_set(
  File "/usr/src/homeassistant/homeassistant/core.py", line 2310, in async_set
    state = State(
            ^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 1788, in __init__
    validate_state(state)
  File "/usr/src/homeassistant/homeassistant/core.py", line 228, in validate_state
    raise InvalidStateError(
homeassistant.exceptions.InvalidStateError: Invalid state with length 543. State max length is 255 characters.
```

I checked met service, and noticed that there was currently 2 warnings... could it be its expecting only 1?

<img width="869" alt="image" src="https://github.com/ciejer/metservice-weather/assets/81659316/4c0b308d-96ea-4bda-8e55-dd45a5954b54">

**Comment by suprra** [comment]
Yep here too, I wonder if you are correct and it's the multiple warnings.  It was working for me a couple of days ago but crapped it self in the last 24

**Comment by NZReDa** [comment]
Same here. Fails to load multiple warnings. Please fix. Thank you.

**homeassistant.exceptions.InvalidStateError: Invalid state with length 327. State max length is 255 characters.**
---
Logger: homeassistant.helpers.entity
Quelle: helpers/entity.py:1205
Erstmals aufgetreten: 28. Juli 2024 um 20:33:11 (74 Vorkommnisse)
Zuletzt protokolliert: 29. Juli 2024 um 20:51:01

Failed to set state for sensor.metservice_nelson_metservice_weather_warnings, fall back to unknown
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1205, in _async_write_ha_state
    hass.states.async_set(
  File "/usr/src/homeassistant/homeassistant/core.py", line 2302, in async_set
    state = State(
            ^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 1773, in __init__
    validate_state(state)
  File "/usr/src/homeassistant/homeassistant/core.py", line 239, in validate_state
    raise InvalidStateError(
homeassistant.exceptions.InvalidStateError: Invalid state with length 327. State max length is 255 characters.

**Comment by Tarawera** [comment]
The message has started to reappear again now after all the foul weather we have been having.

**Comment by ciejer** [comment]
This should be resolved in 0.7 - it truncates the sensor to 255, and the full length message is now available in an attribute of the same sensor. Please advise if any ongoing issues.
Sorry for the delay!
